### PR TITLE
Move commands module into bevy::ecs::world

### PR DESCRIPTION
**Pull request by targrub**
# Objective

Fixes https://github.com/bevyengine/bevy/issues/11628

## Migration Guide

`Command` and `CommandQueue` have migrated from `bevy_ecs::system` to `bevy_ecs::world`, so `use bevy_ecs::world::{Command, CommandQueue};` when necessary.

### CODE REVIEW

File: tools/ci/src/main.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ fn main() {
     if what_to_run.contains(Check::CFG_CHECK) {
         // Check cfg and imports
         std::env::set_var("RUSTFLAGS", "-D warnings");
-        cmd!(sh, "cargo check -Zcheck-cfg --workspace")
+        cmd!(sh, "cargo +nightly check -Zcheck-cfg --workspace")
```
Comment (alice-i-cecile) [inline tools/ci/src/main.rs:164]: This should be its own PR :)

**Comment by alice-i-cecile** [comment]
Definitely in favor of this :) Let me know when it's ready to review.

File: tools/ci/src/main.rs (hunk 1 above)
Comment (targrub) [reply tools/ci/src/main.rs:164]: I want to remove a couple of files from this PR.  What is the correct way to do that?  (In particular, this change to `tools/ci/src/main.rs` and `crates/bevy_macros_compile_fail_tests/tests/deref_mut_derive/missing_deref.fail.stderr`).

Comment (targrub) [reply tools/ci/src/main.rs]: Ok, did it.

**Comment by targrub** [comment]
Ready for review.  @alice-i-cecile @JoJoJet

**Review by alice-i-cecile — APPROVED** [review]
I think this is an improvement.

**Review by JoJoJet — APPROVED** [review]

**Review by james7132 — APPROVED** [review]

File: crates/bevy_ecs/src/system/commands/mod.rs [hunk 2]
```diff
@@ -115,15 +76,6 @@ pub struct Commands<'w, 's> {
...
 }
 
-impl SystemBuffer for CommandQueue {
-    #[inline]
-    fn apply(&mut self, _system_meta: &SystemMeta, world: &mut World) {
-        #[cfg(feature = "trace")]
-        let _span_guard = _system_meta.commands_span.enter();
-        self.apply(world);
-    }
-}
```
Comment (JoJoJet) [inline crates/bevy_ecs/src/system/commands/mod.rs:125]: Nit, but I think it makes sense to keep this impl here, since it's used only for the `Commands` definition just above
